app/views/steps/_form.html.erb

```diff
@@ -1,60 +1,54 @@
-<%
-  left_col = "col-md-7"
-  right_col = "col-md-5"
-%>
+<div class="govuk-grid-row govuk-!-margin-top-7">
+  <div class="govuk-grid-column-two-thirds">
+    <%= render "shared/steps/form_errors", resource: step %>
 
-<%= render "shared/steps/form_errors", resource: step %>
+    <%= render "govuk_publishing_components/components/input", {
+      label: {
+        text: "Step title"
+      },
+      name: "step[title]",
+      value: step.title
+    } %>
 
-<div class="form-group">
-  <div class="row">
-    <div class="<%= left_col %>">
-      <%= form.text_field :title, label: "Step title" %>
-    </div>
-  </div>
-</div>
-
-<div class="form-group">
-  <%= form.label :logic, "Step labelling", default_builder: true %>
-
-  <div class="radio">
-    <label>
-      <%= form.radio_button :logic, "number", default_builder: true, checked: true %>
-      number
-    </label>
-  </div>
-
-  <div class="radio">
-    <label>
-      <%= form.radio_button :logic, "and", default_builder: true %>
-      and
-    </label>
-  </div>
-
-  <div class="radio">
-    <label>
-      <%= form.radio_button :logic, "or", default_builder: true  %>
-      or
-    </label>
-  </div>
-</div>
-
-<div class="form-group">
-  <%= form.label :optional, "Is this step optional?", default_builder: true %>
-
-  <div class="help-block">You must select one of these so that user activity is tracked correctly in Google Analytics.</div>
-
-  <div class="radio">
-    <label>
-      <%= form.radio_button :optional, "false", default_builder: true, checked: true %>
-      essential
-    </label>
-  </div>
+    <%= render "govuk_publishing_components/components/radio", {
+      name: "step[logic]",
+      heading: "Step label",
+      items: [
+        {
+          value: "number",
+          text: "number",
+          checked: !step.logic.present? || step.logic == "number"
+        },
+        {
+          value: "and",
+          text: "and",
+          checked: step.logic == "and"
+        },
+        {
+          value: "or",
+          text: "or",
+          checked: step.logic == "or"
+        }
+      ]
+    } %>
 
-  <div class="radio">
-    <label>
-      <%= form.radio_button :optional, "true", default_builder: true %>
-      optional
-    </label>
+    <%= render "govuk_publishing_components/components/radio", {
+      name: "step[optional]",
+      heading: "Is this step optional?",
+      hint: "You must select one of these so that user activity is tracked correctly in Google Analytics.",
+      items: [
+        {
+          value: "false",
+          text: "essential",
+          checked: !step.optional.present? || !step.optional
```

Comment:
Backwards logic! I think a future enhancement would be to change `!step.optional` to `step.required`.